api/v1alpha1/oidcpolicy_types.go

```diff
@@ -0,0 +1,287 @@
+	gatewayapiv1alpha2 "sigs.k8s.io/gateway-api/apis/v1alpha2"
+)
+
+// EDIT THIS FILE!  THIS IS SCAFFOLDING FOR YOU TO OWN!
+// NOTE: json tags are required.  Any new fields you add must have json tags for the fields to be serialized.
+
+const (
+	DefaultCallbackPath      = "/auth/callback"
+	DefaultTokenExchangePath = "/oauth/token" //nolint:gosec
+	DefaultAuthorizePath     = "/oauth/authorize"
+	DefaultTokenSourceName   = "jwt"
+
+	StatusConditionReady string = "Ready"
+)
+
+// OIDCPolicySpec defines the desired state of OIDCPolicy
+type OIDCPolicySpec struct {
+	// Reference to the object to which this policy applies.
+	// +kubebuilder:validation:XValidation:rule="self.group == 'gateway.networking.k8s.io'",message="Invalid targetRef.group. The only supported value is 'gateway.networking.k8s.io'"
+	// +kubebuilder:validation:XValidation:rule="self.kind == 'HTTPRoute' || self.kind == 'Gateway'",message="Invalid targetRef.kind. The only supported values are 'HTTPRoute' and 'Gateway'"
+	TargetRef gatewayapiv1alpha2.LocalPolicyTargetReferenceWithSectionName `json:"targetRef"`
+
+	// Bare set of policy rules (implicit defaults).
+	OIDCPolicySpecProper `json:""`
+}
+
+// OIDCPolicySpecProper contains common shared fields for the future inclusion of defaults and overrides
+type OIDCPolicySpecProper struct {
+	// Provider holds the information for the OIDC provider
+	Provider *Provider `json:"provider,omitempty"`
+	// Auth holds the information regarding AuthN/AuthZ
+	// +optional
+	Auth *Auth `json:"auth,omitempty"`
+}
+
+type Auth struct {
+	// TokenSource informs where the JWT token will be found in the request for authentication
+	// If omitted, it defaults to credentials passed in cookies.
+	// +optional
+	TokenSource *authorinov1beta3.Credentials `json:"tokenSource,omitempty"`
+	// Claims contains the JWT Claims https://www.rfc-editor.org/rfc/rfc7519.html#section-4
+	// +optional
+	Claims map[string]string `json:"claims,omitempty"`
+}
+
+// Provider defines the settings related to the Identity Provider (IDP)
+type Provider struct {
+	// URL of the OpenID Connect (OIDC) token issuer endpoint.
+	// Use it for automatically discovering the JWKS URL from an OpenID Connect Discovery endpoint (https://openid.net/specs/openid-connect-discovery-1_0.html).
+	// The Well-Known Discovery path (i.e. "/.well-known/openid-configuration") is appended to this URL to fetch the OIDC configuration.
+	IssuerURL string `json:"issuerURL"`
+
+	// Not implemented in AuthPolicy yet
+	// URL of the JSON Web Key Set (JWKS) endpoint.
+	// Use it for non-OpenID Connect (OIDC) JWT authentication, where the JWKS URL is known beforehand.
+	// The JSON Web Keys (JWK) obtained from this endpoint are automatically cached and the caching updated whenever the kid of a JWT does not match any of the cached JWKs (https://openid.net/specs/openid-connect-core-1_0.html#RotateSigKeys).
+	// One of: jwksURL, issuerURL
+	// +optional
+	//JWKSURL string `json:"jwksURL,omitempty"`
```

Comment:
Should we first bump the version of the AuthConfig the Kuadrant Operator depends upon?

Reply:
Yes, that's the idea, so then we can implement it here